build/ci/job-template.yml

```diff
@@ -35,12 +45,32 @@ jobs:
     - ${{ if eq(parameters.pool.name, 'Hosted macOS High Sierra') }}:
       - script: brew update && brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/f5b1ac99a7fba27c19cee0bc4f036775c889b359/Formula/libomp.rb && brew install mono-libgdiplus gettext && brew link gettext --force && brew link libomp --force
         displayName: Install build dependencies
+    - bash: echo "##vso[task.setvariable variable=LD_LIBRARY_PATH]$(nightlyBuildRunPath):$LD_LIBRARY_PATH"
+      displayName: Set LD_LIBRARY_PATH for Ubuntu to locate Native shared library
+      condition: ${{ and(eq(parameters.nightlyBuild, 'true'), eq(parameters.name, 'Ubuntu_x64_NetCoreApp21')) }}
     - script: ${{ parameters.buildScript }} -$(_configuration) -buildArch=${{ parameters.architecture }}
       displayName: Build
+    - script: $(dotnetPath) restore $(nightlyBuildProjPath)
+      displayName: Restore nightly build project
+      condition: eq(${{ parameters.nightlyBuild }}, true)
+    - script: $(dotnetPath) list $(nightlyBuildProjPath) package --source $(nugetFeed) --outdated > $(versionFilePath)
+      displayName: List latest package versions
+      condition: eq(${{ parameters.nightlyBuild }}, true)
+    - script: $(dotnetPath) run --project $(packageUpdaterProjPath)
+      displayName: Update package version
+      condition: eq(${{ parameters.nightlyBuild }}, true)
+    - script: $(dotnetPath) msbuild -restore $(nightlyBuildProjPath) /p:ReferenceTypeForTestFramework="Nuget" /p:Configuration=$(_configuration) /p:TargetArchitecture=${{ parameters.architecture }}
+      displayName: Build Nightly Build Project with latest package versions
+      condition: eq(${{ parameters.nightlyBuild }}, true)
+    - script: ${{ parameters.buildScript }} -$(_configuration) -runnightlybuildtests
+      displayName: Run Nightly Build Tests
+      condition: and(succeeded(), eq(${{ parameters.nightlyBuild }}, true))
     - script: ${{ parameters.buildScript }} -- /t:DownloadExternalTestFiles /p:IncludeBenchmarkData=$(_includeBenchmarkData)
       displayName: Download Benchmark Data
+      condition: eq(${{ parameters.nightlyBuild }}, false)
     - script: ${{ parameters.buildScript }} -$(_configuration) -runtests -coverage=${{ parameters.codeCoverage }}
       displayName: Run Tests.
+      condition: eq(${{ parameters.nightlyBuild }}, false)
```

Comment:
Instead of using runtime conditions, could you use compile time conditions? That way when you're running on nightly build, the UI will only show the "real" steps and when in CI, the "real" CI steps. The way currently is, it will show a bunch of disabled steps, which makes harder to read the logs.

Compile time would be `${{ if eq(parameters.nightlyBuild, 'false') }}`

Reply:
I see, will do, good suggestions

---
In reply to: [344278680](https://github.com/dotnet/machinelearning/pull/4444#discussion_r344278680) [](ancestors = 344278680)